Src/Support/Google.Apis.Auth/OAuth2/ExternalAccountConfiguration.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Holds the configuration for an external account credential.
+    /// </summary>
+    public abstract class ExternalAccountConfiguration
+    {
+        private protected ExternalAccountConfiguration() { }
+
+        /// <summary>
+        /// The STS audience which contains the resource name for the
+        /// workload identity pool or the workforce pool
+        /// and the provider identifier in that pool.
+        /// </summary>
+        public abstract string Audience { get; }
+
+        /// <summary>
+        /// The STS subject token type based on the OAuth 2.0 token exchange spec.
+        /// </summary>
+        public abstract string SubjectTokenType { get; }
+
+        /// <summary>
+        /// The STS token exchange endpoint.
+        /// </summary>
+        public abstract string TokenUrl { get; }
+
+        /// <summary>
+        /// The URL for the service account impersonation request.
+        /// This is only required for workload identity pools when APIs to be accessed
+        /// have not integrated with UberMint. If this is not available,
+        /// the STS returned GCP access token is directly used.
```

Comment:
I'd hyphenate STS-returned for clarity, or maybe "the GCP access token return from STS"?